common/presenters/allocation-people-to-cards.js

```diff
@@ -0,0 +1,26 @@
+const personToCardComponent = require('./person-to-card-component')
+
+function allocationPeopleToCards(moves = []) {
+  const mapper = personToCardComponent()
+  const slotsByFilledStatus = moves.reduce(
+    ([filled, empty], move) => {
+      return move.person
+        ? [[...filled, move], empty]
+        : [filled, [...empty, move]]
+    },
+    [[], []]
+  )
+  const emptySlots = slotsByFilledStatus[1].length
+  const filledSlots = slotsByFilledStatus[0].map(move => {
+    const { person } = move
+    if (!person.fullname) {
+      person.fullname = `${person.first_names} ${person.last_name}`
+    }
+    return mapper(person)
+  })
+  return {
+    emptySlots,
```

Comment:
This is a little hard to read as part of the "presenter". Returning a count does't really fit in with transforming an object into something that needs to be used by a component or another entity. 

I feel this presenter can be simplified a lot but just returning an array of cards for filled slots, and the count of empty slots can be sent to the view via the controller using a filter on the collection:

```javascript
const emptySlots = moves.filter(move => !move.person).length
```

At the moment the name of the presenter is saying it will return cards but it actually returns an object with a count and then a list of cards.

If you remove the count you could even remove the need for this presenter altogether and re-use the existing person-to-card-component:

```javascript
const filledMoves = moves
  .filter(move => move.person)
  .map(move => move.person)
  .map(personToCardComponent())

const locals = {
  ...
  emptySlots,
  filledMoves,
  ...
}
```

This greatly improves its readability and maintainability. If we make changes to how the card displays those changes will be consistently carried across the application.

And if you don't feel moving the addition of `fullname` to use the shared formatter is necessary:

```javascript
const filledMoves = moves
  .filter(move => move.person)
  .map(move => {
    return {
      ...move.person,
      person.fullname: `${person.first_names} ${person.last_name}`,
    }
  })
  .map(personToCardComponent())
```